add MPI example (#778)
kubernetes-sigs/kueue: site/content/en/docs/tasks/run_mpi_jobs.md

---

> title: "Run A MPI Job"

Reviewer: ```suggestion
title: "Run A MPIJob"
```

In fact, it's `MPIJob`.

date: 2023-05-16
weight: 6
description: >

>   Run a Kueue scheduled MPI Job

Reviewer: ```suggestion
  Run a Kueue scheduled MPIJob
```

---

> This page shows how to leverage Kueue's scheduling and resource management capabilities when running [MPI Operator](https://github.com/kubeflow/mpi-operator) jobs.

Reviewer: ```suggestion
This page shows how to leverage Kueue's scheduling and resource management capabilities when running [MPI Operator](https://github.com/kubeflow/mpi-operator) MPIJobs.
```

This guide is for [batch users](/docs/tasks#batch-user) that have a basic understanding of Kueue. For more information, see [Kueue's overview](/docs/overview).

## Before you begin

Check [administer cluster quotas](/docs/tasks/administer_cluster_quotas) for details on the initial cluster setup.

Check [the MPI Operator installation guide](https://github.com/kubeflow/mpi-operator#installation).

## MPI Operator definition

### a. Queue selection

The target [local queue](/docs/concepts/local_queue) should be specified in the `metadata.labels` section of the MPIJob configuration.

```yaml
metadata:
  labels:
    kueue.x-k8s.io/queue-name: user-queue
```

### b. Set Suspend field in MPIOperator

```yaml
spec:
  runPolicy:

>     suspend: true

Reviewer 2: Is this necessary, we'll suspend the batch Job whatever the value of this field by webhook mutation.

Reviewer 3: Yes, this is not necessary, but I think it's still useful to highlight.

Maybe you can say (optional) and explain that kueue will set this to false when the jobs is admitted.

Author: Ah.  I didn't even know that!  I'll update.

```

> ### c. Set MPIJobs as an allowed workload in Kueue Configuration

Reviewer 3: This should be in the "Before you begin" section.

Because not every user needs to do this, only the person configuring Kueue.
Also, link to https://kueue.sigs.k8s.io/docs/installation/#install-a-custom-configured-released-version

Author: Updated.

This guide is for [batch users](/docs/tasks#batch-user) that have a basic understanding of Kueue. For more information, see [Kueue's overview](/docs/overview).

## Before you begin

Check [administer cluster quotas](/docs/tasks/administer_cluster_quotas) for details on the initial cluster setup.

Check [the MPI Operator installation guide](https://github.com/kubeflow/mpi-operator#installation).

Set MPIJobs as an allowed workload in Kueue Configuration.

```yaml
integrations:
  frameworks:
  - "kubeflow.org/mpijob"
```

You can [modify kueue configurations from installed releases](/docs/installation#install-a-custom-configured-released-version) to include MPIJobs as an allowed workload.  

## MPI Operator definition

### a. Queue selection

The target [local queue](/docs/concepts/local_queue) should be specified in the `metadata.labels` section of the MPIJob configuration.

```yaml
metadata:
  labels:
    kueue.x-k8s.io/queue-name: user-queue
```

### b. Optionally set Suspend field in MPIJobs

```yaml
spec:
  runPolicy:
    suspend: true
```

By default, Kueue will set `suspend` to true via webhook and unsuspend it when the MPIJob is admitted.

## Sample MPI Job

> 

Reviewer: I think it is worth mentioning that this example is based on https://github.com/kubeflow/mpi-operator/blob/ccf2756f749336d652fa6b10a732e241a40c7aa6/examples/v2beta1/pi/pi.yaml.

---
title: "Run A MPIJob"
date: 2023-05-16
weight: 6
description: >
  Run a Kueue scheduled MPIJob
---

This page shows how to leverage Kueue's scheduling and resource management capabilities when running [MPI Operator](https://github.com/kubeflow/mpi-operator) MPIJobs.

This guide is for [batch users](/docs/tasks#batch-user) that have a basic understanding of Kueue. For more information, see [Kueue's overview](/docs/overview).

## Before you begin

Check [administer cluster quotas](/docs/tasks/administer_cluster_quotas) for details on the initial cluster setup.

Check [the MPI Operator installation guide](https://github.com/kubeflow/mpi-operator#installation).

Set MPIJobs as an allowed workload in Kueue Configuration.

```yaml
integrations:
  frameworks:

>   - "batch/job"

Reviewer: ```suggestion
```

Users can use MPIjob integration without specifying batch/job.

Author: done.

  - "kubeflow.org/mpijob"
```

You can [modify kueue configurations from installed releases](/docs/installation#install-a-custom-configured-released-version) to include MPIJobs as an allowed workload.  

## MPI Operator definition

### a. Queue selection

The target [local queue](/docs/concepts/local_queue) should be specified in the `metadata.labels` section of the MPIJob configuration.

```yaml
metadata:
  labels:
    kueue.x-k8s.io/queue-name: user-queue
```

> ### b. Optionally set Suspend field in MPIOperator

Reviewer 2: ```suggestion
### b. Optionally set Suspend field in MPIJob
```
right?

Reviewer: right. That's `MPIJob`.

Author: Done.

```yaml
spec:
  runPolicy:
    suspend: true
```

> By default, Kueue will add this field with its webhook.   

Reviewer 2: ```suggestion
By default, Kueue will set `suspend` to true via webhook and unsuspend it when the MPIJob is admitted.
```

Author: done.

---

> title: "Run A MPIJob"

Reviewer 3: ```suggestion
title: "Run an MPIJob"
```
